app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -1,215 +1,229 @@
-import androidx.compose.material3.Scaffold
-import androidx.compose.material3.Text
-import androidx.compose.material3.TextButton
-import androidx.compose.material3.TopAppBar
-import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.runtime.setValue
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.LocalConfiguration
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.text.TextStyle
 import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
-import androidx.lifecycle.viewmodel.compose.viewModel
 import coil.compose.AsyncImage
-import com.github.se.icebreakrr.model.profile.Gender
 import com.github.se.icebreakrr.model.profile.Profile
+import com.github.se.icebreakrr.model.profile.ProfilesViewModel
 import com.github.se.icebreakrr.model.tags.TagsViewModel
 import com.github.se.icebreakrr.ui.navigation.NavigationActions
-import com.github.se.icebreakrr.ui.sections.tagSelectorTextSizeFactor
 import com.github.se.icebreakrr.ui.tags.TagSelector
-import com.google.firebase.Timestamp
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
 
+@SuppressLint("UnrememberedMutableState", "CoroutineCreationDuringComposition")
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ProfileEditingScreen(
     navigationActions: NavigationActions,
-    tagsViewModel: TagsViewModel = viewModel(factory = TagsViewModel.Factory)
+    tagsViewModel: TagsViewModel,
+    profilesViewModel: ProfilesViewModel
 ) {
 
   val configuration = LocalConfiguration.current
   val screenWidth = configuration.screenWidthDp.dp
   val screenHeight = configuration.screenHeightDp.dp
 
-  // Define dynamic padding and text size based on screen size
-  val padding = screenWidth * 0.02f // 2% of screen width
-  val textSize = screenWidth * 0.06f // 6% of screen width
+  val padding = screenWidth * 0.02f
+  val textSize = screenWidth * 0.08f
 
-  // TODO : make the padding and text size dynamic based on screen size
+  val profilePictureSize = screenWidth * 0.25f
+  val catchphraseHeight = screenHeight * 0.10f
+  val descriptionHeight = screenHeight * 0.16f
 
-  val user =
-      Profile(
-          uid = "",
-          name = "",
-          gender = Gender.OTHER,
-          birthDate = Timestamp.now(),
-          catchPhrase = "",
-          description = "",
-          tags = listOf(),
-          profilePictureUrl = "")
+  LaunchedEffect(Unit) {
+    Firebase.auth.currentUser?.let { profilesViewModel.getProfileByUid(it.uid) }
+  }
```

Comment:
Suggestion: now that I think of it, we could directly ad that in the settings screen as it is needed for the setting screen card, then your own profile's preview and only then edit profile. By doing this `getProfileByUid` only once we could reduce the number of network call, load every page (after setting) faster and reduce the probability of net error

Reply:
I agree with you, thus there is already 38 implementations of profilesViewModel.getProfileByUid(it.uid) especially in the tests of the viewmodel and the firebase, thus I'd let the author (Arthur) do it.